src/amber.cc

```diff
@@ -96,6 +98,36 @@ amber::Result Amber::ExecuteWithShaderData(const amber::Recipe* recipe,
         return r;
       }
     }
+
+    if (!dump_buffer && buffer_info.buffer_name == "buffer")
+      dump_buffer = true;
+  }
+
+  if (!dump_buffer)
+    return engine->Shutdown();
+
+  opts->extractions.erase(
```

Comment:
The rest of this method seems overly complicated. When requesting a buffer dump, we should ask for the buffer name to be dumped to be provided on the command line, and give the ability to request multiple buffers. We then just dump that specific buffer. That code can go in the loop above which dumps framebuffer.